frontend/routes/components/dialogContainer/components/revisionsDialog/revisionsDialog.component.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+import DialogActions from '@material-ui/core/DialogActions';
+import Button from '@material-ui/core/Button';
+import { DATE_TIME_FORMAT } from '../../../../../services/formatting/formatDate';
+import { DateTime } from '../../../dateTime/dateTime.component';
+import {
+	Item,
+	Row,
+	Tag,
+	Column,
+	Description,
+	StyledList,
+	Property,
+	PropertyWrapper,
+	StyledDialogContent
+} from './revisionsDialog.styles';
+import { renderWhenTrue } from '../../../../../helpers/rendering';
+
+interface IProps {
+	currentRevisionName: string;
+	currentRevisionId: string;
+	currentModelName: string;
+	revisions: any[];
+	handleSetNewRevision: (revision) => void;
+	handleClose: () => void;
+}
+
+const setNewRevision = (handler, revision, isTheSameRevision) => {
+	if (isTheSameRevision) {
+		return;
+	}
+	handler(revision);
+};
+
+export const RevisionsDialog = (props: IProps) => {
+	const { revisions, handleSetNewRevision, currentRevisionId } = props;
+
+	return (
+		<>
+			<StyledDialogContent>
+				<StyledList>
+					{
+						renderWhenTrue(() => {
```

Comment:
Wow, what happened here?

1. Create a new function which is wrapped by `renderWhentTrue` and returns the rows
2. Create a new function which returns single row